staker/stakerapp.go

```diff
@@ -1626,6 +1623,62 @@ func (app *App) StakeExpand(
 	}
 }
 
+// ConsolidateUTXOs consolidates UTXOs into a single larger UTXO
+// This is a public method that wraps the internal consolidateUTXOs method
+func (app *App) ConsolidateUTXOs(stakerAddress btcutil.Address, targetAmount int64) (*chainhash.Hash, error) {
+	// check we are not shutting down
+	select {
+	case <-app.quit:
+		return nil, nil
+	default:
+	}
+
+	// Create script for the target amount to the staker address
+	changeScript, err := txscript.PayToAddrScript(stakerAddress)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create change script: %w", err)
+	}
+
+	// Create single output for the target amount plus estimated fees
+	outputs := []*wire.TxOut{{
+		Value:    targetAmount,
+		PkScript: changeScript,
+	}}
+
+	feeRate := app.feeEstimator.EstimateFeePerKb()
+
+	// Create the transaction - WalletController will automatically select the best UTXOs
+	tx, err := app.wc.CreateAndSignTx(outputs, btcutil.Amount(feeRate), stakerAddress, app.filterUtxoFnGen())
+	if err != nil {
+		return nil, fmt.Errorf("failed to create consolidation transaction: %w", err)
+	}
+
+	// Send the transaction
+	txHash, err := app.wc.SendRawTransaction(tx, true)
+	if err != nil {
+		return nil, fmt.Errorf("failed to send consolidation transaction: %w", err)
+	}
+
+	app.logger.WithFields(logrus.Fields{
+		"stakerAddress":       stakerAddress,
+		"targetAmount":        targetAmount,
+		"consolidationTxHash": txHash,
+	}).Infof("Successfully consolidated UTXOs")
+
+	_, err = app.notifier.RegisterConfirmationsNtfn(
```

Comment:
this call returns something called `ConfirmationEvent` if we would like to wait for confirmation we would need to do something like (wait on confirmation channel):
```
<-ev.Confirmed
```

Given that this waiting for confirmation can take a long time, enough that RPC command would timeout, I would just remove this registration. It will be up to user to wait for confirmation of the consolidation transaction.